specs/icmp.rflx

```diff
@@ -45,93 +45,74 @@ package ICMP is
    type Uint_32_Unused is range 0 .. 0 with Size => 32;
    type Uint_24_Unused is range 0 .. 0 with Size => 24;
 
-   type Echo_Request_Reply_Message is
+   type Message is
       message
-         Tag : Tag;
-         Code : Code_Zero;
-         Checksum : Checksum;
-         Identifier : Identifier;
-         Sequence_Number : Sequence_Number
-            then Data
-               with Length => Message'Last - Sequence_Number'Last;
-         Data : Opaque;
-      end message;
-
-   type Destination_Unreachable_Message is
-      message
-         Tag : Tag;
-         Code : Code_Destination_Unreachable;
-         Checksum : Checksum;
-         Unused : Uint_32_Unused
+         Tag : Tag
+            then Code_Destination_Unreachable
+               if Tag = Destination_Unreachable,
+            then Code_Time_Exceeded
+               if Tag = Time_Exceeded,
+            then Code_Redirect
+               if Tag = Redirect,
+            then Code_Zero
+               if (Tag = Echo_Reply or
+                  Tag = Echo_Request or
+                  Tag = Source_Quench or
+                  Tag = Parameter_Problem or
+                  Tag = Timestamp_Msg or
+                  Tag = Timestamp_Reply or
+                  Tag = Information_Request or
+                  Tag = Information_Reply);
+         Code_Destination_Unreachable : Code_Destination_Unreachable
+            then Checksum;
+         Code_Time_Exceeded : Code_Time_Exceeded
+            then Checksum;
+         Code_Redirect : Code_Redirect
+            then Checksum;
+         Code_Zero : Code_Zero
+            then Checksum;
+         Checksum : Checksum
+            then Uint_32_Unused
+               if (Tag = Destination_Unreachable or
+                  Tag = Source_Quench or
+                  Tag = Time_Exceeded),
+            then Identifier
+               if (Tag = Echo_Request or
+                  Tag = Echo_Reply or
+                  Tag = Timestamp_Msg or
+                  Tag = Timestamp_Reply or
+                  Tag = Information_Request or
+                  Tag = Information_Reply),
+            then Pointer
+               if Tag = Parameter_Problem,
+            then Gateway_Inet_Addr
+               if Tag = Redirect;
+         Gateway_Inet_Addr : Gateway_Inet_Addr
```

Comment:
I would prefer the use of the same name as in the RFC: Gateway_Internet_Address.

Reply:
changed the name as suggested in bb76642